### PR TITLE
state vector to probability utility method

### LINKED ISSUE #4322: Add `cirq.state_vector_to_probabilities`
**Is your feature request related to a use case or problem? Please describe.**
I need a function that takes in a `state_vector` and returns the probabilities of it's amplitudes.

**Describe the solution you'd like**
Add a `def state_vector_to_probabilities()` to 
https://github.com/quantumlib/Cirq/blob/master/cirq-core/cirq/value/probability.py

**[optional] Describe alternatives/workarounds you've considered**
Defined my own:
```python
from cirq.value.probability import validate_probability
def state_vector_to_probabilities(state_vector: np.ndarray) -> np.ndarray:
      probabilities = np.abs(state_vector) ** 2
      validate_probability(probabilities, "sum of the amplitudes of the state vector")
      return probabilities

```
**What is the urgency from your perspective for this issue? Is it blocking important work?**
P3 - I'm not really blocked by it, it is an idea I'd like to discuss / suggestion based on principle

### PR DESCRIPTION
**Pull request by ishmum123**
Closes #4322

### CODE REVIEW

File: cirq-core/cirq/experiments/cross_entropy_benchmarking.py [hunk 1]
```diff
@@ -405,7 +405,7 @@ def cross_entropy_benchmarking(
         probs_exp_k = []  # type: List[np.ndarray]
         for circ_k in circuits_k:
             res = simulator.simulate(circ_k, qubit_order=qubits)
-            state_probs = np.abs(np.asarray(res.final_state_vector)) ** 2  # type: ignore
+            state_probs = cirq.state_vector_to_probabilities(np.asarray(res.final_state_vector))  # type: ignore
```
Comment (vtomole) [inline cirq-core/cirq/experiments/cross_entropy_benchmarking.py:408]: Does the mypy check pass if we remove this `# type: ignore`?

File: cirq-core/cirq/value/probability.py [hunk 2]
```diff
@@ -13,7 +13,8 @@
 # limitations under the License.
 
 """Utilities for handling probabilities."""
-
+import numpy as np
+from cirq.qis import to_valid_state_vector as format
```
Comment (vtomole) [inline cirq-core/cirq/value/probability.py:17]:
```suggestion
from cirq.qis import to_valid_state_vector
```

File: cirq-core/cirq/value/probability.py [hunk 3]
```diff
@@ -33,3 +34,8 @@ def validate_probability(p: float, p_str: str) -> float:
     elif p > 1:
         raise ValueError(f'{p_str} was greater than 1.')
     return p
+
+
+def state_vector_to_probabilities(state_vector: 'cirq.STATE_VECTOR_LIKE') -> np.ndarray:
+    normalized_vector = format(state_vector)
```
Comment (vtomole) [inline cirq-core/cirq/value/probability.py:40]:
```suggestion
    valid_state_vector = to_valid_state_vector(state_vector)
```

File: cirq-core/cirq/value/probability.py [hunk 4]
```diff
@@ -33,3 +34,8 @@ def validate_probability(p: float, p_str: str) -> float:
...
+    return np.abs(normalized_vector) ** 2
```
Comment (vtomole) [inline cirq-core/cirq/value/probability.py:41]:
```suggestion
    return np.abs(valid_state_vector) ** 2
```

**Review by vtomole — CHANGES_REQUESTED** [review]
Thanks for opening this. I've left some comments and suggestions.

File: cirq-core/cirq/__init__.py [hunk 5]
```diff
@@ -30,9 +30,9 @@
     from cirq import (
         # Core
         protocols,
-        value,
         linalg,
         qis,
+        value,
```
Comment (vtomole) [inline cirq-core/cirq/__init__.py:35]: Why was this moved?

**Review by vtomole — CHANGES_REQUESTED** [review]

File: cirq-core/cirq/__init__.py (hunk 5 above)
Comment (ishmum123) [reply cirq-core/cirq/__init__.py]: One of the tests was failing due to a recursive dependency. Can't find which one, sorry. Reverted now.

File: cirq-core/cirq/experiments/cross_entropy_benchmarking.py (hunk 1 above)
Comment (ishmum123) [reply cirq-core/cirq/experiments/cross_entropy_benchmarking.py]: Yes it does apparently. I have removed it in the new commit

File: cirq-core/cirq/__init__.py [hunk 6]
```diff
@@ -460,6 +460,7 @@
     TParamKey,
     TParamVal,
     validate_probability,
+    state_vector_to_probabilities,
```
Comment (vtomole) [inline cirq-core/cirq/__init__.py:463]: The listing should be alphabetical. Move this to before all the `T`'s

**Review by vtomole — CHANGES_REQUESTED** [review]

File: cirq-core/cirq/value/probability.py [hunk 7]
```diff
@@ -13,6 +13,8 @@
 # limitations under the License.
 
 """Utilities for handling probabilities."""
+import numpy as np
+from cirq.qis import to_valid_state_vector
```
Comment (vtomole) [inline cirq-core/cirq/value/probability.py:17]: Add 
```python
if TYPE_CHECKING:
      import cirq
```

**Review by vtomole — CHANGES_REQUESTED** [review]
run `./check/format-incremental --apply` locally and push the changes.

File: cirq-core/cirq/value/probability.py (hunk 7 above)
Comment (ishmum123) [reply cirq-core/cirq/value/probability.py:17]: I am getting this - `NameError: name 'TYPE_CHECKING' is not defined`

File: cirq-core/cirq/experiments/cross_entropy_benchmarking.py [hunk 8]
```diff
@@ -30,7 +30,7 @@
 import numpy as np
 import scipy
 from matplotlib import pyplot as plt
-from cirq import circuits, devices, ops, protocols, sim, work
+from cirq import circuits, devices, ops, protocols, sim, work, value
```
Comment (vtomole) [inline cirq-core/cirq/experiments/cross_entropy_benchmarking.py:33]: Alphabetical ordering.

```suggestion
from cirq import circuits, devices, ops, protocols, sim, value, work```

File: cirq-core/cirq/value/__init__.py [hunk 9]
```diff
@@ -47,9 +47,7 @@
     MeasurementKey,
 )
 
-from cirq.value.probability import (
-    validate_probability,
-)
+from cirq.value.probability import validate_probability, state_vector_to_probabilities
```
Comment (vtomole) [inline cirq-core/cirq/value/__init__.py:50]: Order alphabetically.

**Review by vtomole — CHANGES_REQUESTED** [review]
This looks good to me. I have a few small suggestions left.

File: cirq-core/cirq/value/__init__.py [hunk 10]
```diff
@@ -47,7 +47,7 @@
     MeasurementKey,
 )
 
-from cirq.value.probability import validate_probability, state_vector_to_probabilities
+from cirq.value.probability import state_vector_to_probabilities, validate_probability
```
Comment (vtomole) [inline cirq-core/cirq/value/__init__.py:50]: For consistency

```suggestion
from cirq.value.probability import (state_vector_to_probabilities, validate_probability)
```

**Review by vtomole — CHANGES_REQUESTED** [review]

**Review by vtomole — APPROVED** [review]
LGTM. Thanks for your patience!

Note: You'll need to run the formatter again to pass the format check.

**Comment by ishmum123** [comment]
> LGTM. Thanks for your patience!
> 
> Note: You'll need to run the formatter again to pass the format check.

I did run it just now, the formatter just removed the parens we just added

**Comment by vtomole** [comment]
Okay.  Let's revert my suggestion.

**Comment by vtomole** [comment]
Looks like there is a bug with the checks. We will merge this PR once https://github.com/quantumlib/Cirq/issues/4448 is fixed.
